cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AbstractNotificationSubscriberService.java

```diff
@@ -223,50 +218,49 @@ public Void call(DatasetContext context) throws Exception {
      * Processes a list of notifications
      *
      * @param context the dataset context
-     * @param notifications the list of fetched notifications from TMS
+     * @param notifications the list of pairs of message ids and fetched notifications from TMS
      * @throws Exception
      */
-    public void fetchAndProcessNotifications(DatasetContext context, List<Notification> notifications)
+    void processFetchedNotifications(DatasetContext context, List<Notification> notifications)
       throws IOException, DatasetManagementException {
+
       for (Notification notification : notifications) {
         processNotification(context, notification);
       }
     }
 
-    protected JobQueueDataset getJobQueue() {
-      return Transactions.executeUnchecked(transactional, new TxCallable<JobQueueDataset>() {
-        @Override
-        public JobQueueDataset call(DatasetContext context) throws Exception {
-          return Schedulers.getJobQueue(context, datasetFramework);
-        }
-      });
-    }
-
-    protected AppMetadataStore getAppMetadataStore(DatasetContext datasetContext) throws IOException,
-      DatasetManagementException {
+    AppMetadataStore getAppMetadataStore(DatasetContext context) {
```

Comment:
It will be used in `ScheduleNotificationSubscriberService` once we implement program status scheduling